### ISSUE TITLE
HELP WANTED: Maintainers/Collaborators/Contributors!

### ISSUE THREAD

**Issue by vector-of-bool** [vector-of-bool/vscode-cmake-tools#315]
It's becoming obvious that the real bottleneck to the project is myself. I have a lot of things I'd like to get done, and this is only one of many hobby projects (The only one so far with any success). I take _far_ too long to respond to issue and PR comments, and I have limited access to resources. Every project has a sweet spot in the ratio between userbase size and developer team size. At the moment, the ~1:10,000 ratio really isn't going to cut it. I need help!

I note several frequent PR and issue openers/commenters who do great work in investigating bugs and fetaures, and it has been suggested at least once in the past that this project could be well served by having more maintainers. (where maintainers == users with write-access to the repository).

To this end, I'd like to reach out and find users who would be willing to share the responsibility for maintaining this tool.

There is so much I'd like to do with this project, and I think it's high-time that it gets a real team. The C and C++ development workflow for VSCode still has a long way to catch up, but I think it can be a formidable contender in the C++ IDE space. I see CMake Tools (among others) being a cornerstone of that goal, and producing a good tool requires good people behind it. I can't do this on my own anymore and haven't been able to for quite a while.

Having a flourishing and appreciative userbase is so rewarding, and I'm so glad I've come this far, but I can't keep going it alone.

I need help with:

- _Feature development_: There's a good stack of feature requests in the backlog that I'd like to get to.
- _Bugfixes_: There are quite a few moving parts, and things break, especially with the many disparate systems that CMake Tools supports.
- _Automated Test Development_: I've neglected to do a lot of work in the automated testing phase. Testing this extension is a bit harder than I expected, but it's completely possible.
- _Continuous Development/Integration_: There is an existing open PR to add CI systems to the extension for proper testing, and I hope that we can see CI merged into the project soon.
- **Documentation**: Even if you have no familiarity with TypeScript, you can help write and maintain the documentation. There's a lot to cover, and any help would be appreciated.
- _Issue and Release Triage_: Having recently passed 100 open issues, I'm starting to feel buried. I tried using releases and milestones previously but fell behind.
- **macOS**: Halp. I don't have one, and all macOS issues just spin in circles until someone else can identify the exact problem or send a PR.

Without further delay: This is my call to action. If you are interested, please **send an email** to [`vectorofbool@gmail.com`](mailto:vectorofbool@gmail.com). Include in the email your GitHub username, area of interest from above, and any other details of particular interest.

Thanks for reading!

**Comment by Randshot** [comment]
This is only somewhat related to the above, but I would suggest that for now, all new PRs should be based on the latest refactor branch, since that is going to be the new structure of the project.
I don't see much sense in committing anything to the dev branch ATM, since it has to be rewritten for the new structure anyway.

On the topic of Maintainers/Collaborators/Contributors, I would be interested, but I can't really promise any consistent input, since I have many other things to do. This would also be pretty much my first project with TypeScript. Just putting that out there.

**Comment by vector-of-bool** [comment]
Thanks @Randshot. Since you're still pretty new to this, I'll wait on giving you write access just yet. Keep up sending PRs and commenting on issues, though: It's a real help.

Don't worry about expectations of consistent contribution. Even _I_ can't maintain consistency in this project, so I wouldn't be expecting that of others.

And don't worry about not knowing TypeScript: This is also my first project in it. I found it a joy to work with after dealing with the treacherous untyped-ness of most scripting languages. Just poke around the codebase and make yourself familiar. Also [check out the dev-docs](https://vector-of-bool.github.io/docs/vscode-cmake-tools/dev/index.html) (Maybe a bit out-of-date).

**Comment by vector-of-bool** [comment]
I'm going to close this up, although anyone stumbling on here is still welcome to email me.
